sql-plugin/src/main/spark340/scala/com/nvidia/spark/rapids/shims/CreateDataSourceTableAsSelectCommandMetaShims.scala

```diff
@@ -48,10 +47,9 @@ final class CreateDataSourceTableAsSelectCommandMeta(
 
     val spark = SparkSession.active
     origProvider =
-      GpuDataSource.lookupDataSourceWithFallback(cmd.table.provider.get, spark.sessionState.conf)
-    // Note that the data source V2 always fallsback to the V1 currently.
-    // If that changes then this will start failing because we don't have a mapping.
-    gpuProvider = origProvider.getConstructor().newInstance() match {
+      GpuDataSourceBase.lookupDataSourceWithFallback(cmd.table.provider.get,
+        spark.sessionState.conf)
+    origProvider.getConstructor().newInstance() match {
       case f: FileFormat if GpuOrcFileFormat.isSparkOrcFormat(f) =>
         GpuOrcFileFormat.tagGpuSupport(this, spark, cmd.table.storage.properties, cmd.query.schema)
       case _: ParquetFileFormat =>
```

Comment:
Nit: There's now an extraneous `None` in the `case ds =>` below since the match result is no longer being used.